Code/Legacy/CryCommon/IAudioInterfacesCommonData.h

```diff
@@ -425,151 +394,51 @@ namespace Audio
-            static SAudioCallBackInfos emptyInstance;
-            return emptyInstance;
+            return (m_owner == rhs.m_owner);
         }
 
-        void* const pObjectToNotify;
-        UserData pUserData;
-        void* const pUserDataOwner;
-        const TATLEnumFlagsType nRequestFlags;
+        inline bool operator!=(const TriggerNotificationIdType& rhs) const
+        {
+            return !(*this == rhs); 
+        }
     };
 
-    ///////////////////////////////////////////////////////////////////////////////////////////////////
-    struct SAudioRequest
-    {
-        SAudioRequest()
-            : nFlags(eARF_NONE)
-            , nAudioObjectID(INVALID_AUDIO_OBJECT_ID)
-            , pOwner(nullptr)
-            , pUserData(nullptr)
-            , pUserDataOwner(nullptr)
-            , pData(nullptr)
-        {}
-
-        ~SAudioRequest() {}
 
-        SAudioRequest(const SAudioRequest& other) = delete;
-        SAudioRequest& operator=(const SAudioRequest& other) = delete;
-
-        TATLEnumFlagsType nFlags;
-        TAudioObjectID nAudioObjectID;
-        void* pOwner;
-        void* pUserData;
-        void* pUserDataOwner;
-        SAudioRequestDataBase* pData;
-    };
+} // namespace Audio
 
-    ///////////////////////////////////////////////////////////////////////////////////////////////////
-    struct SAudioRequestInfo
+namespace AZStd
+{
+    template<>
+    struct hash<Audio::TriggerNotificationIdType>
     {
-        explicit SAudioRequestInfo(
-            const EAudioRequestResult ePassedResult,
-            void* const pPassedOwner,
-            void* const pPassedUserData,
-            void* const pPassedUserDataOwner,
-            const EAudioRequestType ePassedAudioRequestType,
-            const TATLEnumFlagsType nPassedSpecificAudioRequest,
-            const TAudioControlID nPassedAudioControlID,
-            const TAudioObjectID nPassedAudioObjectID,
-            const TAudioEventID passedAudioEventID)
-            : eResult(ePassedResult)
-            , pOwner(pPassedOwner)
-            , pUserData(pPassedUserData)
-            , pUserDataOwner(pPassedUserDataOwner)
-            , eAudioRequestType(ePassedAudioRequestType)
-            , nSpecificAudioRequest(nPassedSpecificAudioRequest)
-            , nAudioControlID(nPassedAudioControlID)
-            , nAudioObjectID(nPassedAudioObjectID)
-            , audioEventID(passedAudioEventID)
-        {}
-
-        const EAudioRequestResult eResult;
-        void* const pOwner;
-        void* const pUserData;
-        void* const pUserDataOwner;
-        const EAudioRequestType eAudioRequestType;
-        const TATLEnumFlagsType nSpecificAudioRequest;
-        const TAudioControlID nAudioControlID;
-        const TAudioObjectID nAudioObjectID;
-        const TAudioEventID audioEventID;
+        using argument_type = Audio::TriggerNotificationIdType;
+        using result_type = size_t;
+        inline result_type operator()(const argument_type& id) const
```

Comment:
argument_type and result_type typedefs for hash are deprecated and will be removed in C++20(https://en.cppreference.com/w/cpp/utility/hash).
The type alias should removed and the actual argument type should be specified with the a return type of size_t.

Also the hash function could be marked constexpr as well.

Reply:
I can remove the typedefs and add `constexpr`